pkg/domain/attributor.go

```diff
@@ -0,0 +1,41 @@
+package domain
+
+import (
+	"context"
+
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/nexpose"
+)
+
+// AssetAttributor receives a NexposeAssetVulnerabilities instance, queries one
+// or more asset inventory systems, and returns a NexposeAttributedAssetVulnerabilities
+// instance annotated with the business context for the asset at scan time.
+type AssetAttributor interface {
+	Attribute(ctx context.Context, asset NexposeAssetVulnerabilities) (NexposeAttributedAssetVulnerabilities, error)
+}
+
+// NexposeAssetVulnerabilities is a Nexpose asset response payload appended
+// with assetVulnerabilityDetails
+type NexposeAssetVulnerabilities struct {
+	Asset           *nexpose.Asset
+	Vulnerabilities []*nexpose.AssetVulnerabilityDetails `json:"assetVulnerabilityDetails"`
+}
+
+// CloudAssetDetails represent an asset and associated metadata
+type CloudAssetDetails struct {
+	PrivateIPAddresses []string
+	PublicIPAddresses  []string
+	Hostnames          []string
+	ResourceType       string
+	AccountID          string
+	Region             string
+	ResourceID         string
+	Tags               map[string]string
+}
+
+// NexposeAttributedAssetVulnerabilities is a NexposeAssetVulnerabilities instance combined
+// with the business context pertaining to the asset at scan time.
+type NexposeAttributedAssetVulnerabilities struct {
```

Comment:
This doesn't seem to match the expected payload from the docs - I understand that these can and will change but how should we go about doing so? Do service PRs drive doc updates or vice versa?